database/optionally.go

```diff
@@ -0,0 +1,130 @@
+package database
+
+import (
+	"context"
+	"fmt"
+	"github.com/icinga/icinga-go-library/com"
+	"github.com/pkg/errors"
+)
+
+// Upsert upserts entities to the database.
```

Comment:
Upsert inserts new rows into a table or updates rows of a table if the primary key already exists.